dig.go

```diff
@@ -372,10 +395,28 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 			return err
 		}
 	}
+	c.addNode(n)
 
 	return nil
 }
 
+// addNode adds the dotEdges in node n into the DOT-format graph in Container c.
+func (c *Container) addNode(n *node) {
+	pnodes := n.paramList.DotNodes()
+	rnodes := n.resultList.DotNodes()
+	edges := make([]dotEdge, len(pnodes)*len(rnodes))
+	i := 0
```

Comment:
Rather than allocating an `i` here, we can use the index from the `range` in our loop:
```
for  i, pnode := range pnodes { ... } // i == 0 -> i == 1, ...
```

Reply:
On second thought, now that we are collecting the `edges` slice and simply expanding it into the `c.dg.edges` slice (via the `c.dg.edges = append(c.dg.edges, edges...)` call below), it's probably easier to append directly into the `c.dg.edges`.

This would look similar to:
```
for _, pnode := range pnodes {
  for _, rnode:= range rnodes {
    c.dg.edges = append(c.dg.edges, dotEdge{param: pnode, result: rnode})
}
```